shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java

```diff
@@ -241,27 +241,28 @@ public void setPaddingTopToZeroForFullscreenMode() {
     assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
 
     // Then we simulate the system applying a window inset.
-    WindowInsets windowInsets = mock(WindowInsets.class);
-    when(windowInsets.getSystemWindowInsetTop()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetBottom()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetLeft()).thenReturn(100);
-    when(windowInsets.getSystemWindowInsetRight()).thenReturn(100);
+    WindowInsets windowInsets =
+        new WindowInsets.Builder()
+            .setInsets(
+                android.view.WindowInsets.Type.navigationBars()
+                    | android.view.WindowInsets.Type.statusBars(),
+                Insets.of(100, 100, 100, 100))
+            .build();
     flutterView.onApplyWindowInsets(windowInsets);
 
     // Verify.
-    verify(flutterRenderer, times(2)).setViewportMetrics(viewportMetricsCaptor.capture());
-    assertEquals(0, viewportMetricsCaptor.getValue().viewPaddingTop);
```

Comment:
interesting. was this a bug in the test?

Reply:
These tests have not been running because the current Robolectric configuration is skipping anything with an API level above 28.  The behavior of the code diverged from the expectations of the test, and AFAICT the code is correct.